borg/testsuite/helpers.py

```diff
@@ -259,6 +259,41 @@ def testInvalidUnicode(self):
         assert e.match(str(b"ba\x80/foo", 'latin1'))
 
 
+@pytest.mark.parametrize("lines, expected", [
+    ([], None),
+    (["# Comment only"], None),
+    (["*"], []),
+    (["# Comment",
+      "*/something00.txt",
+      "  whitespace\t",
+      "/whitespace/at/end of filename \t ",
+      # Whitespace before comment
+      " #/ws*",
+      # Empty line
+      "",
+      "# EOF"],
+     ["/more/data", "/home"]),
+    ])
+@pytest.mark.parametrize("eol", ["\n", "\r", "\r\n"])
+def test_patterns_from_file(tmpdir, lines, eol, expected):
+    files = [
+        '/data/something00.txt', '/more/data', '/home',
+        ' #/wsfoobar',
+        '/whitespace/at/end of filename \t ',
+    ]
+
+    def evaluate(filename):
+        patterns = load_excludes(open(filename, "r"))
+        return [path for path in files if not exclude_path(path, patterns)]
+
+    exclfile = tmpdir.join("exclude.txt")
+
+    with exclfile.open("w") as fh:
```

Comment:
Dropped the EOL part of the test. Introducing binary I/O would've required quite a bit more work.